src/kbmod/reprojection_utils.py

```diff
@@ -217,6 +218,123 @@ def correct_parallax_geometrically(coord, obstime, point_on_earth, heliocentric_
+    # coordinate origins changes the RA, DEC so that its line of sight
+    # now contains the object, which isn't true for ICRS coordinate
+    # Copy the cartesian elements of the LOS ray for use later
+    point_on_earth_geocentric=point_on_earth.geocentric*u.m
+    mjd_times = Time(mjds, format="mjd")
+    los_earth_obj = SkyCoord(
+        ra=ra*u.deg,
+        dec=dec*u.deg,
+        obstime=mjd_times
+    ).transform_to(GCRS(obsgeoloc=point_on_earth_geocentric))
+    los_earth_obj_cart = los_earth_obj.cartesian
+
+    # Now that we have the ray elements, re-create the SkyCoords with but with an
+    # added distance placeholder, because that alters the underlying coordinate
+    # representation to SphericalRepresentation. Avoid data copying
+    los_earth_obj = SkyCoord(
+        ra=los_earth_obj.ra,
+        dec=los_earth_obj.dec,
+        distance=np.zeros((len(ra), ))*u.AU,
+        obstime=los_earth_obj.obstime,
+        obsgeoloc=los_earth_obj.obsgeoloc,
+        frame="gcrs",
+        copy=False
+        )
+
+    # for each ephemeris calculate the geocentric distance
+    # that matches the given heliocentric distance guess and update
+    # our LOS coordinate with these new distances
+    for obstime, ext, eyt, ezt in zip(unique_obstimes,
+                                      location_ephems_x,
+                                      location_ephems_y,
+                                      location_ephems_z):
+        mjd_mask = mjds == obstime
+        vx = los_earth_obj_cart.x[mjd_mask].value
+        vy = los_earth_obj_cart.y[mjd_mask].value
+        vz = los_earth_obj_cart.z[mjd_mask].value
+
+        # Solve the quadratic equation for the ray leaving the earth and intersecting
+        # a sphere centered on the barycenter at (0, 0, 0) with radius = heliocentric_distance
+        a = vx * vx + vy * vy + vz * vz
+        b = 2 * vx * ext + 2 * vy * eyt + 2 * vz * ezt
+        c = ext * ext + eyt * eyt + ezt * ezt - heliocentric_distance * heliocentric_distance
+        disc = b * b - 4 * a * c
+
+        invalid_result_mask = disc < 0
+        disc[disc<0] = 0
+
+        # Since the ray will be starting from within the sphere (we assume the
+        # heliocentric_distance is at least 1 au), one of the solutions should be positive
+        # and the other negative. We only use the positive one.
+        dist = (-b + np.sqrt(disc)) / (2 * a)
+        # we can't set the distance to negative in SkyCOord without a lot of wrangling
+        # the distance of 0 should correspond to UnitSphericalRepresentation
+        dist[invalid_result_mask] = 0.0
+        los_earth_obj.distance[mjd_mask] = dist*u.AU
+        break
+
+    # finally, transform the coordinates with the new distance guesses
+    # back to ICRS, which will now include the correct parallax correction
+    if return_geo_dists:
+        return los_earth_obj.transform_to(ICRS()), los_earth_obj.distance
+    return los_earth_obj.transform_to(ICRS())
```

Comment:
what is the situation where we don't return the fits? maybe a quick comment here

Reply:
Returning the geocentric distance to the object is superfluous, the coordinates should now contain the guess distance to the object so calculating them is trivial, but storing them costs a lot (for 4.7 mil rows f.e). I wasn't sure why we return those, but we don't need them. 

I left them in so people don't have to break their script, but generally they are not supposed to be required...
Added some kind of a comment to this point but didn't try to figure this out.